pkg/jira/issue_test.go

```diff
@@ -554,3 +554,33 @@ func TestRemoteLinkIssue(t *testing.T) {
 	err = client.RemoteLinkIssue("TEST-1", "weblink title", "https://weblink.com")
 	assert.Error(t, &ErrUnexpectedResponse{}, err)
 }
+
+func TestWatchIssue(t *testing.T) {
+	var unexpectedStatusCode bool
+
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		assert.Equal(t, "POST", r.Method)
+		assert.Equal(t, "application/json", r.Header.Get("Accept"))
+		assert.Equal(t, "application/json", r.Header.Get("Content-Type"))
+
+		assert.Equal(t, "/rest/api/3/issue/TEST-1/watchers", r.URL.Path)
```

Comment:
Let's make sure we are calling v2 version of the Jira API

```suggestion
		assert.Equal(t, "/rest/api/2/issue/TEST-1/watchers", r.URL.Path)
```